server/client.go

```diff
@@ -3443,10 +3514,19 @@ func (c *client) deliverMsg(prodIsMQTT bool, sub *subscription, acc *Account, su
 
 	// Check for closed connection
 	if client.isClosed() {
+		if mt != nil {
```

Comment:
If function call not too heavy hit on performance (need to measure) we could add check for receiver being nil in addEvent functions to clean up a bit.

Reply:
Seems to be ok. I did the ok, I ran the whole bench publish tests, but since it will be more prevalent when there are subscriptions (multiple egresses), here is an excerpt:
```
                                      │  new1.txt   │             new2.txt              │
                                      │   sec/op    │   sec/op     vs base              │
Publish/MsgSz=0b/Subs=1xAsync-8         200.7n ± 1%   200.1n ± 1%       ~ (p=0.416 n=6)
Publish/MsgSz=0b/Subs=1xAsyncQueue-8    209.6n ± 0%   210.2n ± 0%       ~ (p=0.065 n=6)
Publish/MsgSz=0b/Subs=10xAsync-8        880.2n ± 6%   885.2n ± 6%       ~ (p=0.818 n=6)
Publish/MsgSz=0b/Subs=10xAsyncQueue-8   264.0n ± 4%   265.4n ± 2%       ~ (p=0.669 n=6)
geomean                                 314.4n        315.3n       +0.28%
```